tck/src/main/java/ee/jakarta/tck/concurrent/spec/Platform/virtual/VirtualThreadServlet.java

```diff
@@ -0,0 +1,468 @@
+    
+            Set<Thread> virtualThreads = new HashSet<Thread>();
+    
+            if (result0 instanceof Throwable)
+                throw new AssertionError("An error occured on thread.", (Throwable) result0);
+            assertTrue(isVirtual((Thread) thread0));
+            virtualThreads.add((Thread) thread0);
+    
+            if (result1 instanceof Throwable)
+                throw new AssertionError("An error occured on thread.", (Throwable) result1);
+            assertTrue(isVirtual((Thread) thread1));
+            virtualThreads.add((Thread) thread1);
+    
+            if (result2 instanceof Throwable)
+                throw new AssertionError("An error occured on thread.", (Throwable) result2);
+            assertTrue(isVirtual((Thread) thread2));
+            virtualThreads.add((Thread) thread2);
+    
+            assertEquals(3, virtualThreads.size(), "Each task execution should have used a different virtual thread");
+        });
+    }
+
+    public void testPlatformThreadFactory() throws Exception {
+        ManagedThreadFactory platfromThreadFactoryAnno = InitialContext
+                .doLookup("java:app/concurrent/ThreadFactoryAnnoPlatform");
+        ManagedThreadFactory platfromThreadFactoryDD = InitialContext
+                .doLookup("java:app/concurrent/ThreadFactoryDDPlatform");
+        
+        assertNotNull(platfromThreadFactoryAnno);
+        assertNotNull(platfromThreadFactoryDD);
+
+        assertFalse(isVirtual(platfromThreadFactoryAnno.newThread(NOOP_RUNNABLE)),
+                "Thread Factory should not have returned a virtual thread when defined with virtual=false");
+        assertFalse(isVirtual(platfromThreadFactoryDD.newThread(NOOP_RUNNABLE)),
+                "Thread Factory should not have returned a virtual thread when defined with virtual=false");
+    }
+
+    public void testVirtualThreadFactory() throws Exception {
+        ManagedThreadFactory virtualThreadFactoryAnno = InitialContext
+                .doLookup("java:app/concurrent/ThreadFactoryAnnoVirtual");
+        ManagedThreadFactory virtualThreadFactoryDD = InitialContext
+                .doLookup("java:app/concurrent/ThreadFactoryDDVirtual");
+        
+        assertNotNull(virtualThreadFactoryAnno);
+        assertNotNull(virtualThreadFactoryDD);
+        
+        Thread annoThread = virtualThreadFactoryAnno.newThread(NOOP_RUNNABLE);
+        Thread ddThread = virtualThreadFactoryDD.newThread(NOOP_RUNNABLE);
+        
+        if (VERSION == 17) { //TODO remove when Concurrency API supports only 21+
+            assertThrows(NoSuchMethodException.class, () -> isVirtual(annoThread), "Should be impossible to get a virtual thread on Java 17");
+            assertThrows(NoSuchMethodException.class, () -> isVirtual(ddThread), "Should be impossible to get a virtual thread on Java 17");
+            return;
+        }
+
+        // Java 21+
+        assertEquals(isVirtual(annoThread), isVirtual(ddThread));
```

Comment:
Similar to first comment. This is not a requirement, but very unlikely it would not be the case.